src/core/installAddon.js

```diff
@@ -221,48 +163,33 @@ type FindInstallUrlParams = {|
  * platform.
  */
 export const findInstallURL = ({
+  _findFileForPlatform = findFileForPlatform,
   appendSource = true,
   defaultInstallSource,
   location,
   platformFiles,
   userAgentInfo,
 }: FindInstallUrlParams): string | void => {
-  if (!platformFiles) {
-    throw new Error('The platformFiles parameter is required');
-  }
-  if (!userAgentInfo) {
-    throw new Error('The userAgentInfo parameter is required');
-  }
-
   let source;
   if (appendSource) {
-    if (!location) {
-      throw new Error(
-        'The location parameter is required when appendSource is true',
-      );
-    }
+    invariant(
+      location,
+      'The location parameter is required when appendSource is true',
+    );
     source = location.query.src || defaultInstallSource;
   }
 
-  const agentOsName =
-    userAgentInfo.os.name && userAgentInfo.os.name.toLowerCase();
-  const platform = agentOsName && userAgentOSToPlatform[agentOsName];
-  const platformFile = platform && platformFiles[platform];
-
-  let installURL;
-  if (platformFile) {
-    installURL = platformFile.url;
-  }
+  const platformFile = _findFileForPlatform({
+    platformFiles,
+    userAgentInfo,
+  });
 
-  if (!installURL && platformFiles[OS_ALL]) {
-    installURL = platformFiles[OS_ALL].url;
-  }
+  const installURL = platformFile && platformFile.url;
 
   if (!installURL) {
     // This could happen for themes which do not have version files.
     log.debug(
-      oneLine`No install URL exists for platform "${agentOsName}"
-      (mapped to "${platform}"); platform files:`,
+      oneLine` No file exists for os "${userAgentInfo.os.toString()}"; platform files:`,
```

Comment:
Any reason why there's a space at the beginning? It doesn't really matter, just wondering.

Reply:
No reason. I'm not sure how that got there, but it's a mistake. I'll fix it.